### PR TITLE
[Stats Refresh] add structure for stats detail list view

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #n/a

This add the table structure to display stats' detail list views.

- When `View more` is selected on a stat card, the new `SiteStatsDetailTableViewController` is displayed.
  - Note: Custom `View more` are _not_ included (i.e. Latest Post Summary and Posting Activity.)
- For any `View more` selection, temporary data is displayed for now.
- On the list view:
  - The view title matches the selected stat title.
  - The column headers match the selected stat column headers.
  
Note: there's an extra line below the detail list view. That is specific to each stat type configuration. I'll address that when I add support for each type.

To test:

--- 
- On any applicable stat card, select `View more`.
- Verify:
  - The title is the stat type.
  - The column headers are specific to the stat type.
  - There are 3 fake data rows.

detail_labels

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    private typealias Style = WPStyleGuide.Stats
+
+    // MARK: - Init
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // TODO: customize title per stat type
+        title = "This Be The Deets"
```
Comment (stevebaranski) [inline WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift]: 😂

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import UIKit
+
+class SiteStatsDetailTableViewController: UITableViewController, StoryboardLoadable {
+
+    // MARK: - StoryboardLoadable Protocol
+
+    static var defaultStoryboardName = "SiteStatsDetailTableViewController"
```
Comment (stevebaranski) [inline WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift]: Based on how the `StoryboardLoadable` protocol extension works, it should be possible to use the following if you are so inclined:
```
static var defaultStoryboardName = defaultControllerID
```

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    private weak var siteStatsInsightsDelegate: SiteStatsInsightsDelegate?
+    private weak var siteStatsPeriodDelegate: SiteStatsPeriodDelegate?
+
+    private lazy var tableHandler: ImmuTableViewHandler = {
+        return ImmuTableViewHandler(takeOver: self)
+    }()
+
+    // MARK: - View
+
+    override func viewDidLoad() {
```
Comment (stevebaranski) [inline WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift]: Perhaps we could delete this method until we need to add some additional behavior?

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift (hunk 2 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift]: Oooh, I never noticed that. And it does in fact work. Thanks!

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift (hunk 3 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift]: Good catch! It shall be gone!

**Comment by ScoutHarris** [comment]
Thanks @stevebaranski !

**Pull request merged**